app/src/main/java/org/covidwatch/android/ui/onboarding/OnboardingFragment.kt

```diff
@@ -0,0 +1,55 @@
+package org.covidwatch.android.ui.onboarding
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import androidx.navigation.fragment.findNavController
+import androidx.viewpager2.widget.ViewPager2
+import com.google.android.material.tabs.TabLayoutMediator
+import org.covidwatch.android.R
+import org.covidwatch.android.data.UserFlowRepository
+import org.covidwatch.android.databinding.FragmentOnboardingBinding
+import org.covidwatch.android.ui.BaseFragment
+import org.koin.android.ext.android.inject
+
+class OnboardingFragment : BaseFragment<FragmentOnboardingBinding>() {
+
+    private val userFlowRepository: UserFlowRepository by inject()
+    private lateinit var pagerAdapter: OnboardingPagerAdapter
+
+    private val onPageChangeCallback = object : ViewPager2.OnPageChangeCallback() {
+        override fun onPageSelected(position: Int) {
+            super.onPageSelected(position)
+            val isLastPage = position == pagerAdapter.itemCount - 1
+            binding.tabLayout.isVisible = !isLastPage
+            binding.continueSetupButton.isVisible = isLastPage
+        }
+    }
+
+    override fun binding(
+        inflater: LayoutInflater,
+        container: ViewGroup?
+    ): FragmentOnboardingBinding = FragmentOnboardingBinding.inflate(inflater, container, false)
```

Comment:
What's the advantage of binding like this?

Reply:
binding is abstract function of `BaseFragment`. the advantage is that we don't have to write `onCreateView()` `onDestroyView()` methods ourselves